app/repository/sql/slug_lookup.sql

```diff
@@ -1,20 +1,33 @@
-SELECT
-    slug.family_document_import_id, slug.family_import_id
+-- First query for family document slugs
+SELECT DISTINCT
+    slug.family_document_import_id,
+    slug.family_import_id
 FROM slug
-LEFT JOIN family ON family.import_id = slug.family_import_id
-LEFT JOIN family_corpus ON family_corpus.family_import_id = family.import_id
-LEFT JOIN corpus ON corpus.import_id = family_corpus.corpus_import_id
-WHERE slug.name = '{slug_name}'
-AND corpus.import_id IN ({allowed_corpora_ids})
+    INNER JOIN family_document
+        ON slug.family_document_import_id = family_document.import_id
+    INNER JOIN family
+        ON family_document.family_import_id = family.import_id
+    INNER JOIN family_corpus
+        ON family.import_id = family_corpus.family_import_id
+    INNER JOIN corpus
+        ON family_corpus.corpus_import_id = corpus.import_id
+WHERE
+    slug.name = :slug_name
+    AND corpus.import_id = ANY(:allowed_corpora_ids)
 
 UNION
 
-SELECT
-    slug.family_document_import_id, slug.family_import_id
+-- Second query for family slugs
+SELECT DISTINCT
+    NULL AS family_document_import_id,
```

Comment:
Just for my own understanding, what is the point of setting family_document_import_id to null here?

Reply:
In the query, we're setting NULL for the family document import ID as that second query is getting family info only whereas the first query gets family document info. Because we're doing a UNION, we need to return the same fields from both queries :)